src/frontend/src/catalog/system_catalog/pg_catalog/pg_matviews.rs

```diff
@@ -12,80 +12,41 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::LazyLock;
+
 use risingwave_common::catalog::PG_CATALOG_SCHEMA_NAME;
-use risingwave_common::error::Result;
-use risingwave_common::row::OwnedRow;
-use risingwave_common::types::{DataType, ScalarImpl};
-use serde_json::json;
+use risingwave_common::types::DataType;
 
-use crate::catalog::system_catalog::{BuiltinTable, SysCatalogReaderImpl};
+use crate::catalog::system_catalog::BuiltinView;
 
 /// The view `pg_matviews` provides access to useful information about each materialized view in the
 /// database.
 /// Ref: [`https://www.postgresql.org/docs/current/view-pg-matviews.html`]
-pub const PG_MATVIEWS: BuiltinTable = BuiltinTable {
+pub static PG_MATVIEWS: LazyLock<BuiltinView> = LazyLock::new(|| BuiltinView {
     name: "pg_matviews",
     schema: PG_CATALOG_SCHEMA_NAME,
     columns: &[
         (DataType::Varchar, "schemaname"),
         (DataType::Varchar, "matviewname"),
         (DataType::Int32, "matviewowner"),
-        // TODO: add index back when open create index doc again.
-        // (DataType::Boolean, "hasindexes"),
         (DataType::Varchar, "definition"),
         // Below are some columns that PostgreSQL doesn't have.
+        // TODO: these field is only exist in RW and used by cloud side, need to remove it and let
+        // cloud switch to use `rw_catalog.rw_relation_info`.
         (DataType::Int32, "matviewid"),
         (DataType::Varchar, "matviewtimezone"), /* The timezone used to interpret ambiguous
                                                  * dates/timestamps as tstz */
         (DataType::Varchar, "matviewgraph"), /* materialized view graph is json encoded fragment
                                               * infos. */
     ],
-    pk: &[0, 1],
```

Comment:
Note there is currently no pk on `rw_relation_info`, which means there would also be no pk derived for `pg_matviews`. Shall we fix `rw_relation_info`?

Reply:
Good catch, I will fix it.